lektor_tags.py

```diff
@@ -187,6 +187,9 @@ def on_setup_env(self, **extra):
         self.env.jinja_env.loader.searchpath.append(pkg_dir)
         self.env.add_build_program(TagPage, TagPageBuildProgram)
 
+        # Add the `tagweights` dictionary to the jinja environment
+        self.env.jinja_env.globals["tagweights"] = TagWeights(self)
```

Comment:
This should be useless, since the very same line is present in the `on_before_build_all()` method, but `on_before_build_all()` is not always called before the very first build with `lektor server`. Should I report a bug on [lektor](http://github.com/lektor/lektor)?

Reply:
Doesn't necessarily sound like a bug. Do you need the `on_before_build_all` hook?

Reply:
Yes. If it is not used, when using `lektor server`, TagWeights object is created in `on_setup_env()`, and is *never* updated (i.e. tag are counted once, even if tags are added/removed, they are not counted again). With `on_before_build_all()`, they are counted again at each build (but only once if they are needed several times in the same build).

Reply:
I see, it's probably good to say that with a comment then. Like `# Needed to invalidate the tag cache before each build`

Reply:
I assume it is not called for the first build with lektor server if there is nothing to be built? If there are other cases it does sound like a bug.